CaloMC/src/CaloShowerROMaker_module.cc

```diff
@@ -226,9 +243,10 @@ namespace mu2e {
           {
               const CaloShowerStep& step = *istep;
 
-              // time folding and filtering, see docdb-3425 for a stunning explanation
+              // Time folding and filtering, see doc-db 3425 + talks from D. Brown for explanation
               double hitTimeUnfolded = toff_.totalTimeOffset(istep->simParticle())+istep->time();
-              double hitTime         = fmod(hitTimeUnfolded,mbtime_);
+              double hitTime         = hitTimeUnfolded + pbtmc.pbtime_;
+              if (ewMarker.spillType() == EventWindowMarker::SpillType::onspill) hitTime = fmod(hitTime+mbtime_,mbtime_);
 
               if (hitTime < blindTime_ || hitTime > mbtime_ ) continue;
```

Comment:
It looks like you're limiting digitization during the offspill to only the first 1695 ns. Also a few other modules like CaloDigiMaker similarly seem to use mbtime to set the waveform size

Reply:
Crap x2, I forgot this was still wrong in the tracker code as well - you shouldn't use eventLength for this in CaloDigiMaker since the front end will not know this information. you should instead define a condition like blindTime that is the end of your digitization window (for onspill only) and for offspill ignore it. I'll make a PR for the tracker change so you can see what I mean.

Reply:
Hopefully I fixed it. If you tell me how to test it, I'll do it.

Reply:
Ok, tell em when you have it figured out and I'll do the same.